### PR TITLE
SQL: Upgrade H2GIS library to 1.5.0

### PR DESCRIPTION
**Pull request by imotov**
Upgrades H2GIS to 1.5.0, restores H2 version to 1.4.197 and removes
work-around for bugs in H2 1.4.196 and jar conflicts in H2GIS 1.3.2.

### CODE REVIEW

File: x-pack/plugin/sql/qa/build.gradle [hunk 1]
```diff
@@ -16,8 +16,12 @@ dependencies {
 
   // CLI testing dependencies
   compile project(path: xpackModule('sql:sql-cli'), configuration: 'nodeps')
-  compile "org.orbisgis:h2gis-ext:1.3.2"
- 
+  
+  // H2GIS testing dependencies
+  compile ("org.orbisgis:h2gis:1.5.0") {
```
Comment (costin) [inline x-pack/plugin/sql/qa/build.gradle]: I opt for externalizing h2gis (and h2 while at it) versions into a central place (like `jlineVersion), namely `build.gradle` under `sql`: https://github.com/elastic/elasticsearch/blob/f36d6a96e4b38a2a25eacd4692870ef93e1d4a3e/x-pack/plugin/sql/build.gradle#L13

**Review by costin — APPROVED** [review]
LGTM

**Review by matriv — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/qa/build.gradle (hunk 1 above)
Comment (imotov) [reply x-pack/plugin/sql/qa/build.gradle]: :+1: I will do that in master first and then make corresponding changes in the branch after merging

**Review by astefan — APPROVED** [review]
LGTM

**Review by alpar-t — APPROVED** [review]

File: x-pack/plugin/sql/qa/build.gradle (hunk 1 above)
Comment (imotov) [reply x-pack/plugin/sql/qa/build.gradle]: I opened #40551 to address this in master. I will fix it for h2gis when I merge master into geosql branch.

**Pull request merged**